Demo/EncoderFactoryExtension.cs

```diff
@@ -0,0 +1,31 @@
+﻿using Encoder;
+using Encoder.Factory;
+using Encoder.Source;
+using System;
+
+namespace Demo
+{
+    public static class EncoderFactoryExtension
+    {
+        public static object GetSource(this char sourceType)
+        {
+            return sourceType switch
+            {
+                'a' => new Audio(),
+                'v' => new Video(),
+                'i' => new Random().Next(1, 100),
+                _ => new Audio(),
+            };
+        }
+        public static BaseEncoder GetEncoder(this char encoderType, object source)
+        {
+            return encoderType switch
+            {
+                'a' => new AudioFactory().GetEncoder(source),
+                'v' => new VideoFactory().GetEncoder(source),
+                'i' => new IntegerFactory().GetEncoder(source),
+                _ => new AudioFactory().GetEncoder(source),
+            };
```

Comment:
As I see these extension methods are used only once and possibly they will not be used again in other place. 
that's why I think it is better to move this into private methods rather than creating extension methods.